# Profitable App Profiles for the App Store and Google Play Markets
This project aims to find out what types of apps are likely to attract more users on Google Play and App Store and, therefore, are more profitable. The analysis is made just for English apps that are free to dowload and install, so the main source of revenue consists of in-app ads. It means that the revenue for any given app is mostly influenced by the number of users who use the app.

In 2019, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play. These are two data sets that seem suitable for our purpose:

* A data set containing data about approximately ten thousand Android apps from Google Play
* A data set containing data about approximately seven thousand iOS apps from the App Store

Let's start with opening both data sets and then we'll explore them.

In [1]:
# App Store data set
open_file_1 = open('AppleStore.csv', encoding="utf8")
from csv import reader
read_file_1 = reader (open_file_1)
apple_store = list(read_file_1)

In [2]:
# Google Play data set
open_file_2 = open('googleplaystore.csv', encoding="utf8")
read_file_2 = reader (open_file_2)
android = list(read_file_2)

## Exploring Data Sets

In [3]:
# function for exploring data (to look at few rows choosing `start` and `end` indexes)
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [4]:
explore_data (apple_store, 0, 5, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


Number of rows: 7198
Number of columns: 17


In [5]:
explore_data (android, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


#### Apple iOS Store data set has:
* A header
* Number of rows: 7198
* Number of columns: 17

#### Google Play Store data set has:
* A header
* Number of rows: 10842
* Number of columns: 13

## Columns Examination

Detailed explanation of all columns can be found at [iOS](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) and [Google](https://www.kaggle.com/lava18/google-play-store-apps/home)

We already can see some useful columns

|iOS column name        | GooglePlay column name    | Description                       |
|-----------------------|---------------------------|-----------------------------------|
|"track_name"           | "App"                     | App Name                          |
|"size_bytes"           | "Size"                    | Size (in Bytes)                   |
|"price"                | "Price"                   | Price amount                      |
|"prime_genre"          | "Genres"                  | Genre                             |
|"user_rating_ver"      | "Rating"                  | Average User Rating               |
|"cont_rating"          | "Content Rating"          | Content Rating                    |

In total, iOS list has 17 columns and Google Play list - 13:

In [6]:
print("iOS Store:",apple_store[0])
print("Total:", len(apple_store[0]), "\n")

print("Google Play Store:", android[0])
print("Total:", len(android[0]))

iOS Store: ['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
Total: 17 

Google Play Store: ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
Total: 13


## Data Cleaning
* errors
* dublicates
* non-english apps
* non-free apps

### Removing Wrong Data 
From this [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) we can see that there is an error in the row `[10473]` of the Google Play data set. Let's check this row and remove it, if necessary.

In [7]:
print(android[0])
print("\n")
print(android[10473])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In 3rd column we can see the rating equals to `19`, which should be less or equal to `5.0`. So let's delete this row to avoid confusions.

In [8]:
del android[10473]              # Do not run more than once!

In [9]:
# let's check, if it was deleted (we expect the next app 'osmino Wi-Fi: free WiFi' to be in the index 10473
# instead of 'Life Made WI-Fi Touchscreen Photo Frame'):
print(android[10473])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


### Deleting Dublicate Data
After examination of data, we found a lot of dublicate data in Google Play Apps data set. 

In [10]:
# For example, there are 2 entries of Facebook and 4 of Instagram
for app in android:
    if app[0] == "Facebook" or app[0] == "Instagram":
        print (app)

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2

In [11]:
# Let's create 2 list: for unique entries and dublicates
unique = []
dublicates = []

for app in android:
    if app[0] in unique:
        dublicates.append(app[0])
    else:
        unique.append(app[0])
        
print ("Number of dublicates is", len(dublicates))
print ("Number of unique apps is", (len(unique)-1)) # -1 because 1st row is a header

Number of dublicates is 1181
Number of unique apps is 9659


   By looking at the example of Facebook and Instagram dublicates, we could see that the only difference among entries was the 4th column that is a "number of users' reviews". It means that the data was collected at different time and higher number means more recent data. In order to keep the most recent information, we will keep entries with the highest numbers of reviews and remove the rest dublicates. 

   In order to do this, we will:
* create a dictionary where keys are unique app names and values are the highest numbers of reviews.
* create a new data set using the dictionary mentioned above.

In [12]:
# creating the dictionary
max_reviews = {}

for app in android[1:]:
    name = app[0]
    n_reviews = int(app[3])
    
    # adding apps and reviews count to our dictionary
    if name not in max_reviews:
        max_reviews[name] = n_reviews
    
    # replacing values with the highest numbers of reviews
    elif name in max_reviews and n_reviews > max_reviews[name]:
        max_reviews[name] = n_reviews

# to check if everything is correct, the lenth of dictionary should be 9659 (number of unique apps):
len(max_reviews)

9659

In [13]:
android_new = []  
added_apps = []   

for app in android[1:]:
    name = app[0]
    n_reviews = int(app[3])
    
    if n_reviews == max_reviews[name] and name not in added_apps:
        android_new.append(app)
        added_apps.append(name)

# checking new dataset. There should be 9659 columns (without any dublicate and without a header)
explore_data(android_new, 0, 5, True)    

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


In [14]:
# to check for dublicates in iOS data set
unique = []
dublicates = []

for app in apple_store:
    if app[2] in unique:
        dublicates.append(app[2])
    else:
        unique.append(app[2])
               
print ("Number of dublicates is", len(dublicates))

Number of dublicates is 2


We found 2 dublicates, but after investigating and reading [this discussion](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion/90409) we found out that there are 2 apps called Mannequin Challenge and 2 apps called VR Roller Coaster that are actually 4 different applications with the same names but from different developers. So we will keep them all.

### Removing Non-English Apps

As our investigation is for English-speaking audience, it makes total sense to remove non-English entries from both sets.
One way to do this is to remove each app with a name containing a symbol that is not commonly used in English text. According to the ASCII (American Standard Code for Information Interchange) system the numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127. 

Below there is a function to check whether a string contains English characters. If a string has more than 3 non-English character, the function returns False, otherwise it returns True. We leave strings with 1-3 non-English characters because some of the apps contain emojis and characters like ™ that fall outside the ASCII range. 

In [15]:
def English_alphabet(string):
    non_eng = 0
    for character in string:
        if ord(character) > 127:
            non_eng += 1
            
    if non_eng > 3:  
        return False
    else:
        return True

In [16]:
android_eng = []
for app in android_new:
    if English_alphabet(app[0]) == True:
        android_eng.append(app)       
        
explore_data(android_eng, 506, 510, True)

['Hily: Dating, Chat, Match, Meet & Hook up', 'DATING', '4.1', '2560', '56M', '100,000+', 'Free', '0', 'Mature 17+', 'Dating', 'August 1, 2018', '2.5.2', '4.1 and up']


['Find Lover', 'DATING', 'NaN', '21', '4.7M', '1,000+', 'Free', '0', 'Mature 17+', 'Dating', 'July 27, 2018', '1.0.2', '4.0.3 and up']


['Free Dating & Flirt Chat - Choice of Love', 'DATING', '4.0', '42197', 'Varies with device', '5,000,000+', 'Free', '0', 'Mature 17+', 'Dating', 'July 24, 2018', 'Varies with device', 'Varies with device']


['muzmatch: Muslim & Arab Singles, Marriage & Dating', 'DATING', '3.8', '5168', '25M', '100,000+', 'Free', '0', 'Mature 17+', 'Dating', 'July 24, 2018', '3.8.75a', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13


In [17]:
apple_eng = []
for app in apple_store[1:]:         # without a header
    if English_alphabet(app[2]):
        apple_eng.append(app)
        
explore_data(apple_eng, 0, 3, True)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 6183
Number of columns: 17


As we can see, we're left with 9614 Android and 6183 iOS apps.

### Removing non-free Apps

This is our last step in cleaning data. As we are analysing free apps, we should isolate them from non-free apps.

In [18]:
android_free = []
for app in android_eng:
    price = app[7]
    if price == '0':
        android_free.append(app)
        
explore_data(android_free, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


In [19]:
apple_free = []
for app in apple_eng:
    price = app[5]
    if price == '0':
        apple_free.append(app)
        
explore_data(apple_free, 15, 18, True)

['31', '288113403', 'iTranslate - Language Translator & Dictionary', '287933440', 'USD', '0', '123215', '25', '3.5', '5', '10.5.4', '4+', 'Productivity', '37', '5', '23', '1']


['34', '288429040', 'LinkedIn', '273844224', 'USD', '0', '71856', '62', '3.5', '4.5', '9.1.32', '4+', 'Social Networking', '37', '2', '23', '1']


['37', '289523017', 'Blackjack by MobilityWare', '105431040', 'USD', '0', '180087', '1101', '3.5', '4.5', '5.5.3', '12+', 'Games', '37', '5', '5', '1']


Number of rows: 3222
Number of columns: 17


We're left with 8864 Android apps and 3222 iOS apps.

## The Most Common Genres


As we've mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because the revenue is highly influenced by the number of people using the apps.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the `prime_genre` column of the App Store data set, and the `Genres` and `Category` columns of the Google Play data set.

We'll build two functions that we can use to analyze the frequency tables:

* One function to generate frequency tables that show percentages
* Another function we can use to display the percentages in a descending order

The `freq_table` function you see below:

* Takes in two parameters: `dataset` and `index`. `Dataset` is expected to be a list of lists, and `index` is expected to be an integer.
* Creates a dictionary `frequency_table`.
* Iterates over the dataset counting frequency of all values as percentage.

The `display_table()`:

* Takes in 2 parameters: `dataset` and `index`. 
* Generates a frequency table using the `freq_table()` function.
* Transforms the frequency table into a list of tuples, then sorts the list in a descending order.
* Prints the entries of the frequency table in descending order.

In [20]:
def freq_table(dataset, index):
    frequency_table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in frequency_table:
            frequency_table[value] += 1
        else:
            frequency_table[value] = 1
            
    table_percentages = {}
    for key in frequency_table:
        percentage = (frequency_table[key] / total) * 100
        table_percentages[key] = percentage 
         
    return table_percentages

In [21]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        if entry[0] > 0.5:                                  
            print(entry[1], ':', entry[0])

In [22]:
# analysing apps in Apple Store 
display_table(apple_free, 12)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008


As we can see, 58,16% of free English apps in Apple Store are Games. Entertainment apps are close to 8%, followed by Photo and Video apps, which are close to 5%. Only 3.66% of the apps are designed for Education, followed by Social Networking apps which amount for 3.29% of the apps in our data set. 

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. 

In [23]:
# analysing CATEGORIES of apps in Google Play Store
display_table(android_free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [24]:
# GENRES in Google Play Store
display_table(android_free, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Analysing `Category` column in Google Play, we can see that there are not so many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the `Genres` column.

What about diiference between these 2 columns, we could notice that `Genres` in Google Play are much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the `Category` column.

The general impression is that most of the free English apps are designed for entertainment (games, photo and video, social networking, sports, music) in Apple Store while Google Play shows a more balanced landscape of both practical and fun apps.

## Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:

* Isolate the apps of each genre.
* Sum up the user ratings for the apps of that genre.
* Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

In [25]:
# Generating a frequency table for the prime_genre column to get the unique app genres
prime_genres = freq_table (apple_free, 12)

In [26]:
# Looping over the unique genres of the App Store data set
for genre in prime_genres:
    total = 0                   # This variable will store the sum of user ratings specific to each genre.
    len_genre = 0               # This variable will store the number of apps specific to each genre.
    
    for app in apple_free:
        genre_app = app[12]
        if genre_app == genre:
            n_ratings = float(app[6])
            total += n_ratings
            len_genre += 1
            
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Productivity : 21028.410714285714
Weather : 52279.892857142855
Shopping : 26919.690476190477
Reference : 74942.11111111111
Finance : 31467.944444444445
Music : 57326.530303030304
Utilities : 18684.456790123455
Travel : 28243.8
Social Networking : 71548.34905660378
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Games : 22788.6696905016
Food & Drink : 33333.92307692308
News : 21248.023255813954
Book : 39758.5
Photo & Video : 28441.54375
Entertainment : 14029.830708661417
Business : 7491.117647058823
Lifestyle : 16485.764705882353
Education : 7003.983050847458
Navigation : 86090.33333333333
Medical : 612.0
Catalogs : 4004.0


It seems like the most popular categories (with more than 30 000 ratings) are: Navigation, Business, Reference, Social Networking, Weather, Finance, Music, Food & Drinks and Book. Let's investigate a bit more.

In [27]:
for app in apple_free:
    if app[12] == 'Navigation':
        print(app[2], ':', app[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


In [28]:
for app in apple_free:
    if app[12] == 'Business':
        print(app[2], ':', app[6])

Indeed Job Search : 38681
AirWatch Agent : 1150
OPEN Forum : 200
Cisco AnyConnect : 825
SayHi Translate : 8623
ADP Mobile Solutions : 8324
Flashlight ◎ : 24744
Adobe Acrobat Reader: View, Create, & Convert PDFs : 20069
DingTalk : 40
Pulse Secure : 53
GreenVPN - Free & fast VPN with unlimited traffic : 464
iPlum Business Phone Number for Calling & Texting : 392
Mon Espace - Pôle emploi : 11
Sideline - 2nd Phone Number : 7907
Scanner App - PDF Document Scan : 11696
Uber Driver : 3289
VPN Go - Safe Fast & Stable VPN Proxy : 881


In [29]:
for app in apple_free:
    if app[12] == 'Reference':
        print(app[2], ':', app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


In [30]:
# because there are too many apps in this category, we'll display 50 random apps as an example
num = 0
print('50 Random Social Networking Apps:\n')       
for app in apple_free:
    if (app[12] == 'Social Networking') & (num < 50):
        print(app[2], ':', app[6])
        num += 1

50 Random Social Networking Apps:

Facebook : 2974676
LinkedIn : 71856
Skype for iPhone : 373519
Tumblr : 334293
Match™ - #1 Dating App. : 60659
WhatsApp Messenger : 287589
TextNow - Unlimited Text + Calls : 164963
Grindr - Gay and same sex guys chat, meet and date : 23201
imo video calls and chat : 18841
Ameba : 269
Weibo : 7265
Badoo - Meet New People, Chat, Socialize. : 34428
Kik : 260965
Qzone : 1649
Fake-A-Location Free ™ : 354
Tango - Free Video Call, Voice and Chat : 75412
MeetMe - Chat and Meet New People : 97072
SimSimi : 23530
Viber Messenger – Text & Call : 164249
Find My Family, Friends & iPhone - Life360 Locator : 43877
Weibo HD : 16772
POF - Best Dating App for Conversations : 52642
GroupMe : 28260
Lobi : 36
WeChat : 34584
ooVoo – Free Video Call, Text and Voice : 177501
Pinterest : 1061624
知乎 : 397
Qzone HD : 458
Skype for iPad : 60163
LINE : 11437
QQ : 9109
LOVOO - Dating Chat : 1985
QQ HD : 5058
Messenger : 351466
eHarmony™ Dating App - Meet Singles : 11124
YouNow: Liv

In [31]:
for app in apple_free:
    if app[12] == 'Weather':
        print(app[2], ':', app[6])

WeatherBug - Local Weather, Radar, Maps, Alerts : 188583
The Weather Channel: Forecast, Radar & Alerts : 495626
AccuWeather - Weather for Life : 144214
MyRadar NOAA Weather Radar Forecast : 150158
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking : 208648
Météo-France : 24
Yurekuru Call : 53
QuakeFeed Earthquake Map, Alerts, and News : 6081
Moji Weather - Free Weather Forecast : 2333
FEMA : 128
Weather Underground: Custom Forecast & Local Radar : 49192
JaxReady : 22
Hurricane Tracker WESH 2 Orlando, Central Florida : 203
Hurricane by American Red Cross : 1158
Weather & Radar : 37
WRAL Weather Alert : 25
Yahoo Weather : 112603
Weather Live Free - Weather Forecast & Alerts : 35702
NOAA Weather Radar - Weather Forecast & HD Radar : 45696
iWeather - World weather forecast : 80
Almanac Long-Range Weather Forecast : 12
TodayAir : 0
Weather - Radar - Storm with Morecast App : 78
Storm Radar : 22792
WarnWetter : 0
wetter.com : 0
Forecast Bar : 375
Freddy the

In [32]:
num = 0
print('50 Random Music Apps:\n')                 #display 50 random apps as an example
for app in apple_free:
    if (app[12] == 'Music') & (num < 50):
        print(app[2], ':', app[6])
        num += 1

50 Random Music Apps:

Pandora - Music & Radio : 1126879
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
Deezer - Listen to your Favorite Music & Playlists : 4677
Sonos Controller : 48905
NRJ Radio : 38
radio.de - Der Radioplayer : 64
Spotify Music : 878563
SoundCloud - Music & Audio : 135744
Sing Karaoke Songs Unlimited with StarMaker : 26227
SoundHound Song Search & Music Player : 82602
Ringtones for iPhone & Ringtone Maker : 25403
Coach Guitar - Lessons & Easy Tabs For Beginners : 2416
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Magic Piano by Smule : 131695
QQ音乐HD : 224
The Singing Machine Mobile Karaoke App : 130
Bandsintown Concerts : 30845
PetitLyrics : 0
edjing Mix:DJ turntable to remix and scratch music : 13580
Smule Sing! : 119316
Amazon Music : 106235
AutoRap by Smule : 18202
My Mixtapez Music : 26286
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
Karaoke - Sing Karaoke, Unlimited Songs! : 286

In [33]:
for app in apple_free:
    if app[12] == 'Book':
        print(app[2], ':', app[6])

Kindle – Read eBooks, Magazines & Textbooks : 252076
OverDrive – Library eBooks and Audiobooks : 65450
Audible – audio books, original series & podcasts : 105274
BookShout: Read eBooks & Track Your Reading Goals : 879
ikouhoushi : 0
Dr. Seuss Treasury — 50 best kids books : 451
Weirdwood Manor : 197
Green Riding Hood : 392
HOOKED - Chat Stories : 47829
Color Therapy Adult Coloring Book for Adults : 84062
MangaTiara - love comic reader : 0
MangaZERO - comic reader : 9
謎解き2016 : 0
謎解き : 0


In [34]:
num = 0
print('50 Random Game Apps:\n')           
for app in apple_free:
    if (app[12] == 'Games') & (num < 50):
        print(app[2], ':', app[6])
        num += 1

50 Random Game Apps:

Hangman. : 42316
Blackjack by MobilityWare : 180087
PAC-MAN : 508808
Beer Pong Game : 187315
NYTimes Crossword - Daily Word Puzzle Game : 53465
Sky Burger - Build & Match Food Free : 114096
Pixel Starships™ : 8Bit MMORPG : 5565
Angry Birds : 824451
Glow Hockey 2 FREE : 186653
Solitaire : 679055
Angry Birds HD : 89110
▻Sudoku : 359832
Virtual Regatta Offshore : 209
Jenga : 32527
Spider Solitaire Free by MobilityWare : 128739
Angry Birds Seasons HD : 75714
Fruit Ninja® : 327025
Temple Run : 1724546
Angry Birds Rio : 170843
Angry Birds Rio HD : 70122
My Horse : 293857
Tiny Tower - Free City Building : 414803
ROBLOX : 183621
Nyan Cat: Lost In Space : 76392
Fishing Kings Free+ : 35058
Werewolf "Nightmare in Prison" : 78
Magic Jigsaw Puzzles : 187666
DragonVale : 503230
Racing Penguin Free - Top Flying and Diving Game : 141224
Phoenix HD : 13106
Sonic CD : 16358
White Tiles 2 : Piano Master ( Don't Hit The White Tiles 4 ) - Free Music Game : 60
Doodle Jump FREE - BE WAR

As we can see, in Navigation category the result was influenced mostly by Waze and Google Maps. The same is in Business and Reference categories: there are few succesful apps which influence the average numbers. These categories might seem more popular than they are. The situation is a bit better in Music category but there is still strong influence by Spotify, Pandora, Shazam, iHeartRadio who are hard to compete against.

The Gaming category is represented by numerous apps, that shows popularity but, at the same time, tough competition in this area. So let's choose something different.

A category of Social Networking is heavily influenced by Facebook, Skype and other giants. And another issue is that many other apps are still turning around that popular giants (for example, Followers - Social Analytics For Instagram, InsTrack for Instagram - Analytics Plus More, FollowMeter for Instagram).

There are also types of apps that are connected with different challenges and issues:
* Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

* Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, while we just want to create an app.

* Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

But Photo and Video category seems interesting: it has different popular apps, not just popular giants but also medium sector. Competion is not so saturated as in games category. We don't need any phical product or service for the app as in food or finance app. We can see apps in this category below:

In [35]:
num = 0
print('50 Random Photo & Video Apps:\n')      
for app in apple_free:
    if (app[12] == 'Photo & Video') & (num < 50):
        print(app[2], ':', app[6])
        num += 1

50 Random Photo & Video Apps:

iSwap Faces LITE : 39722
Shutterfly: Prints, Photo Books, Cards Made Easy : 51427
Epson iPrint : 2838
Photo Transfer App - Easy backup of photos+videos : 15654
Instagram : 2161558
Splice - Video Editor + Movie Maker by GoPro : 28189
Meitu : 6478
Digital Domain : 102
Snapseed : 8683
Kwai - Share your video moments : 668
Photo Lab: Picture Editor, effects & fun face app : 34585
Camera360 - Selfie Filter Camera, Photo Editor : 16729
Snapchat : 323905
Pic Collage - Picture Editor & Photo Collage Maker : 123433
FotoRus -Camera & Photo Editor & Pic Collage Maker : 32558
Meitu HD : 2150
Perfect Image - Pic Collage Maker, Add Text to Photo, Cool Picture Editor : 1646
Visage makeup editor plus photo teeth whitener : 5767
NightShooting : 9
Pic Jointer – Photo Collage, Camera Effects Editor : 51330
Flipagram : 79905
Bazaart Photo Editor Pro and Picture Collage Maker : 4909
LINE Camera - Photo editor, Animated Stamp, Filter : 3978
You Doodle - draw on photos & pictur

## Most Popular Apps by Genre on Google Play

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. 

However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [36]:
display_table(android_free, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043


We don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are. To perform computations, however, we'll need to convert each install number from string to float. This means we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

In [37]:
android_categories = freq_table (android_free, 1)

for category in android_categories:
    total = 0
    len_category = 0
    
    for app in android_free:
        category_app = app[1]
        
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
            
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

According to number of installations, top categories are: 
* communication (more than 38.4 millions of apps installations)
* video_players (> 24.7 millions)
* social (> 23.2 millions)
* photography (> 17.8 millions) 
* productivity (> 16.7 millions)
* game (> 15.5 millions) 
* travel_and_local (> 13.9 millions) 
* entertainment (> 11.6 millions)

Let's examine in details.

In [38]:
# the most popular communication apps
for app in android_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

On average, Communication apps have the most installs: 38,456,119. But this number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 500 million installs. 

In [39]:
for app in android_free:
    if app[1] == 'VIDEO_PLAYERS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


Social category seems popular but it is mostly dominated by Facebook, Instagram, Tumblr and similar.

In [40]:
for app in android_free:
    if app[1] == 'SOCIAL' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Pinterest : 100,000,000+
Google+ : 1,000,000,000+
Badoo - Free Chat & Dating App : 100,000,000+
Tango - Live Video Broadcast : 100,000,000+
Instagram : 1,000,000,000+
Snapchat : 500,000,000+
LinkedIn : 100,000,000+
Tik Tok - including musical.ly : 100,000,000+
BIGO LIVE - Live Stream : 100,000,000+
VK : 100,000,000+


In [41]:
num = 0
print('50 Random Social Networking Apps:\n')                #display 50 random apps as an example
for app in android_free:
    if (app[1] == 'SOCIAL') & (num < 50):
        print(app[0], ':', app[5])
        num += 1

50 Random Social Networking Apps:

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Social network all in one 2018 : 100,000+
Pinterest : 100,000,000+
TextNow - free text + calls : 10,000,000+
Google+ : 1,000,000,000+
The Messenger App : 1,000,000+
Messenger Pro : 1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus : 1,000,000+
Telegram X : 5,000,000+
The Video Messenger App : 100,000+
Jodel - The Hyperlocal App : 1,000,000+
Hide Something - Photo, Video : 5,000,000+
Love Sticker : 1,000,000+
Web Browser & Fast Explorer : 5,000,000+
LiveMe - Video chat, new friends, and make money : 10,000,000+
VidStatus app - Status Videos & Status Downloader : 5,000,000+
Love Images : 1,000,000+
Web Browser ( Fast & Secure Web Explorer) : 500,000+
SPARK - Live random video chat & meet new people : 5,000,000+
Golden telegram : 50,000+
Facebook Local : 1,000,000+
Meet – Talk to Strangers Using Random Video Chat : 5,000,000+
MobilePatrol Public Safety App : 1,000,00

In Google Play dataset we don't have categry Photo and Video, as in App Store, but we have 2 categories apart: Video Player (that we've checked few cell before) and Photography:

In [42]:
num = 0
print('50 Random Photography Apps:\n')        
for app in android_free:
    if (app[1] == 'PHOTOGRAPHY') & (num < 50):
        print(app[0], ':', app[5])
        num += 1

50 Random Photography Apps:

TouchNote: Cards & Gifts : 1,000,000+
FreePrints – Free Photos Delivered : 1,000,000+
Groovebook Photo Books & Gifts : 500,000+
Moony Lab - Print Photos, Books & Magnets ™ : 50,000+
LALALAB prints your photos, photobooks and magnets : 1,000,000+
Snapfish : 1,000,000+
Motorola Camera : 50,000,000+
HD Camera - Best Cam with filters & panorama : 5,000,000+
LightX Photo Editor & Photo Effects : 10,000,000+
Sweet Snap - live filter, Selfie photo edit : 10,000,000+
HD Camera - Quick Snap Photo & Video : 1,000,000+
B612 - Beauty & Filter Camera : 100,000,000+
Waterfall Photo Frames : 1,000,000+
Photo frame : 100,000+
Huji Cam : 5,000,000+
Unicorn Photo : 1,000,000+
HD Camera : 5,000,000+
Makeup Editor -Beauty Photo Editor & Selfie Camera : 1,000,000+
Makeup Photo Editor: Makeup Camera & Makeup Editor : 1,000,000+
Moto Photo Editor : 5,000,000+
InstaBeauty -Makeup Selfie Cam : 50,000,000+
Garden Photo Frames - Garden Photo Editor : 500,000+
Photo Frame : 10,000,000

In [43]:
for app in android_free:
    if app[1] == 'PHOTOGRAPHY' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

B612 - Beauty & Filter Camera : 100,000,000+
YouCam Makeup - Magic Selfie Makeovers : 100,000,000+
Sweet Selfie - selfie camera, beauty cam, photo edit : 100,000,000+
Google Photos : 1,000,000,000+
Retrica : 100,000,000+
Photo Editor Pro : 100,000,000+
BeautyPlus - Easy Photo Editor & Selfie Camera : 100,000,000+
PicsArt Photo Studio: Collage Maker & Pic Editor : 100,000,000+
Photo Collage Editor : 100,000,000+
Z Camera - Photo Editor, Beauty Selfie, Collage : 100,000,000+
PhotoGrid: Video & Pic Collage Maker, Photo Editor : 100,000,000+
Candy Camera - selfie, beauty camera, photo editor : 100,000,000+
YouCam Perfect - Selfie Photo Editor : 100,000,000+
Camera360: Selfie Photo Editor with Funny Sticker : 100,000,000+
S Photo Editor - Collage Maker , Photo Collage : 100,000,000+
AR effect : 100,000,000+
Cymera Camera- Photo Editor, Filter,Collage,Layout : 100,000,000+
LINE Camera - Photo editor : 100,000,000+
Photo Editor Collage Maker Pro : 100,000,000+


In [44]:
for app in android_free:
    if app[1] == 'PHOTOGRAPHY' and (app[5] == '50,000,000+'
                                      or app[5] == '10,000,000+'
                                      or app[5] == '5,000,000+'):
        print(app[0], ':', app[5])

Motorola Camera : 50,000,000+
HD Camera - Best Cam with filters & panorama : 5,000,000+
LightX Photo Editor & Photo Effects : 10,000,000+
Sweet Snap - live filter, Selfie photo edit : 10,000,000+
Huji Cam : 5,000,000+
HD Camera : 5,000,000+
Moto Photo Editor : 5,000,000+
InstaBeauty -Makeup Selfie Cam : 50,000,000+
Photo Frame : 10,000,000+
Selfie Camera - Photo Editor & Filter & Sticker : 50,000,000+
ASUS Gallery : 50,000,000+
Pretty Makeup, Beauty Photo Editor & Snappy Camera : 5,000,000+
Photo Collage - Layout Editor : 10,000,000+
Silent Camera [High Quality] : 5,000,000+
Sweet Camera - Selfie Filters, Beauty Camera : 10,000,000+
Photo Frames : 10,000,000+
Candy selfie - photo editor, live filter camera : 10,000,000+
Open Camera : 10,000,000+
Blur Image Background Editor (Blur Photo Editor) : 5,000,000+
Muzy - Share photos & collages : 5,000,000+
QuickPic - Photo Gallery with Google Drive Support : 10,000,000+
Flickr : 10,000,000+
Camera for Android : 10,000,000+
Photo Mixer : 10,00

As we see, there is just one app that has more than 1,000,000,000+ installs (Google Photos) and some apps with 100,000,000+ installs (less than 20 apps) and quite wide variaty of apps from 5,000,000 to 100,000,000 installs. We could aim for this sector with a photo editor and filter app.

## Conclusions

We analyzed apps on Google Play and App Store by categories / genres and found out which are more popular and, therefore, more profitable.

Our conclusion is that Photo editing Apps are popular, profitable and comfortable to create and maintain. Photo apps are one of the leaders on both Google Play Store and App Store. They have better possibilities for entering the market than Social or Gaming apps. And they don't require phisical product or service as Food, Finance or Shopping Apps. 